packages/gestalt-datepicker/src/DateRange/InternalDatePicker.tsx

```diff
@@ -55,6 +57,42 @@ const InternalDatePickerWithForwardRef = forwardRef<HTMLInputElement, ModifiedPr
       }
     }, [localeData]);
 
+    function getDatesArray(startDate: Date, endDate: Date) {
+      // Ensure the input dates are valid Date objects
+      const start = new Date(startDate);
+      const end = new Date(endDate);
+
+      // Check if the start date is after the end date
+      if (start > end) {
+        return [];
+      }
+
+      const dates = [];
+      const currentDate = start;
+
+      // Loop through all dates from start to end
+      while (currentDate <= end) {
+        // Push the current date to the array
+        dates.push(new Date(currentDate));
+        // Increment the current date by one day
+        currentDate.setDate(currentDate.getDate() + 1);
+      }
+
+      return dates;
+    }
+
+    function generateHighligths(
+      startDate: Date | null | undefined,
+      endDate: Date | null | undefined,
+    ) {
+      const datesArray = startDate && endDate ? getDatesArray(startDate, endDate) : [];
+      return [
+        {
+          'react-datepicker__day--in-range-secondary': datesArray,
```

Comment:
You might be able to pull in styles like:
```
import styles from 'daterange.css`

 {`[styles.days-in-secondary]`:dateArray}
```

Reply:
Can you also add a snapshot test that says `secondaryHighlight range appears when selected` so we can have good coverage

Reply:
sure 👍🏾